gk/main.c

```diff
@@ -417,21 +425,141 @@ setup_gk_instance(unsigned int lcore_id, struct gk_config *gk_conf)
 		goto flow_hash;
 	}
 
+	ret = snprintf(ring_name, sizeof(ring_name), "gk_ring_%d", block_idx);
```

Comment:
As we discussed it before, I used ring in the same way as Cody deals with the ARP cache. In addition, we may only need a few more lines of code to wrap them as library functions. Should we do that?